errors/doc.go

```diff
@@ -0,0 +1,17 @@
+/**
+ Package errors implements custom error interfaces for weave.
+
```

Comment:
Good start.

It would be nice to mention the importance of the abci error codes, as they are trasnmitted to the app and the only machine parse able response on an error